specs/DefaultDownloadDialog.md

```diff
@@ -0,0 +1,221 @@
+{
+  /// `TRUE` if the default download dialog is currently visible.
+  [propget] HRESULT IsDefaultDownloadDialogVisible(
+      [out, retval] BOOL* isDefaultDownloadDialogVisible);
+
+  /// Sets the `IsDefaultDownloadDialogVisible` property. A download is
+  /// considered recent if it was downloaded in this instance of the webview.
+  /// If there has not yet been a recent download, setting the default download
+  /// dialog to visible displays all past downloads for the current profile.
+  /// Otherwise, the dialog will display only the recent downloads with a
+  /// "See more" button, which displays the past downloads when clicked on.
+  ///
+  /// \snippet ViewComponent.cpp ToggleDefaultDownloadDialog
+  [propput] HRESULT IsDefaultDownloadDialogVisible(
+      [in] BOOL isDefaultDownloadDialogVisible);
+
+  /// Adds an event handler for the `DefaultDownloadDialogLayout` event. The event is
+  /// raised when the default download dialog's position needs to be recomputed,
+  /// such as when the dialog is first opened, when the webview is resized, and
+  /// when the webview is navigated. Use the `Position` property on the event
+  /// args to specify the new position. To set the position relative to the
+  /// current webview bounds, use the `LocalBounds` property on the event args.
+  ///
+  /// \snippet ViewComponent.cpp SetDefaultDownloadDialogPosition
+  HRESULT add_DefaultDownloadDialogLayout(
+      [in] ICoreWebView2DefaultDownloadDialogLayoutEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_DefaultDownloadDialogLayout`.
+  HRESULT remove_DefaultDownloadDialogLayout([in] EventRegistrationToken token);
+}
+
+/// Implements the interface to receive `DefaultDownloadDialogLayout` events.
+[uuid(86557854-794a-414a-b046-ba515f617306), object, pointer_default(unique)]
+interface ICoreWebView2DefaultDownloadDialogLayoutEventHandler : IUnknown {
+
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke([in] ICoreWebView2* sender,
+                 [in] ICoreWebView2DefaultDownloadDialogLayoutEventArgs* args);
+}
+
+[uuid(e7d2bdea-2e67-4550-b286-6e7d20653d4e), object, pointer_default(unique)]
+interface ICoreWebView2DefaultDownloadDialogLayoutEventArgs : IUnknown {
+  /// The default download dialog position of the top-left corner, relative to
+  /// the webview bounds. Use `LocalBounds` to check the current webview bounds.
+  /// The default position of the download dialog is calculated with the
+  /// following formula:
+  /// (LocalBounds.right - Width - 30, LocalBounds.top + 15).
+  [propget] HRESULT Position([out, retval] POINT* position);
+
+  /// Sets the default download dialog position. Returns `E_INVALIDARG` if an
+  /// invalid position is provided, and the default position is used instead.
+  [propput] HRESULT Position([in] POINT position);
+
+  /// Get the current webview bounds, with the top-left corner being (0, 0).
```

Comment:
For all of these we should say what the coordinate space is or what the units are.